deps/src/main/java/com/microsoft/azure/sdk/iot/deps/serializer/ErrorCodeDescription.java

```diff
@@ -0,0 +1,193 @@
+package com.microsoft.azure.sdk.iot.deps.serializer;
+
+/**
+ * See https://docs.microsoft.com/en-us/rest/api/iothub/common-error-codes for additional details
+ */
+public enum ErrorCodeDescription
+{
+    // These descriptions belong to fully qualified status codes, such as 404001
+    UnclassifiedErrorCode,
+    InvalidProtocolVersion,
+    DeviceInvalidResultCount,
+    InvalidOperation,
+    ArgumentInvalid,
+    ArgumentNull,
+    IotHubFormatError,
+    DeviceStorageEntitySerializationError,
+    BlobContainerValidationError,
+    ImportWarningExistsError,
+    InvalidSchemaVersion,
+    DeviceDefinedMultipleTimes,
+    DeserializationError,
+    BulkRegistryOperationFailure,
+    CannotRegisterModuleToModule,
+    IotHubNotFound,
+    IotHubUnauthorizedAccess,
+    IotHubUnauthorized,
+    IotHubSuspended,
+    IotHubQuotaExceeded,
+    JobQuotaExceeded,
+    DeviceMaximumQueueDepthExceeded,
+    IotHubMaxCbsTokenExceeded,
+    DeviceNotFound,
+    JobNotFound,
+    PartitionNotFound,
+    ModuleNotFound,
+    DeviceAlreadyExists,
+    ModuleAlreadyExistsOnDevice,
+    DeviceMessageLockLost,
+    MessageTooLarge,
+    TooManyDevices,
+    TooManyModulesOnDevice,
+    ThrottleBacklogLimitExceeded,
+    InvalidThrottleParameter,
+    ServerError,
+    JobCancelled,
+    ConnectionForcefullyClosedOnNewConnection,
+    DeviceNotOnline,
+    DeviceConnectionClosedRemotely,
+
+    // These descriptions belong to 3 digit http response codes such as 404.
+    BadFormat,
+    Unauthorized,
+    Forbidden,
+    NotFound,
+    Conflict,
+
+    /**
+     * Represents status code for 429 and for status code 429001
+     */
+    PreconditionFailed,
+
+    RequestEntityTooLarge,
```

Comment:
Does this comment mean PreconditionFailed maps to 429 and RequestEntityTooLarge to 429001?

Reply:
The javadoc is only visible when using intellisense for the PreconditionFailed enum value. It means that the service may return a 429, and it may return a 429001, but in both cases it maps to this enum value

Reply:
Ah, the spacing and block comments confused me.